server/templates.go

```diff
@@ -0,0 +1,241 @@
+package server
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/bouk/httprouter"
+	"github.com/influxdata/chronograf"
+	"github.com/influxdata/chronograf/uuid"
+)
+
+// ValidTemplateRequest checks if the request sent to the server is the correct format.
+func ValidTemplateRequest(template *chronograf.Template) error {
+	switch template.Type {
+	default:
+		return fmt.Errorf("Unknown template type %s", template.Type)
+	case "query", "constant", "csv", "fieldKeys", "tagKeys", "tagValues":
```

Comment:
We also have databases and measurements as types.  Are those useful to be added?